x-pack/plugins/fleet/server/services/files/client_to_host.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import assert from 'assert';
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import type { Logger } from '@kbn/logging';
+
+import { createEsFileClient, createFileHashTransform } from '@kbn/files-plugin/server';
+
+import { v4 as uuidV4 } from 'uuid';
+
+import { getFileDataIndexName, getFileMetadataIndexName } from '../../../common';
+
+import { FleetFilesClientError } from '../../errors';
+
+import type { FleetFile, FleetToHostFileClientInterface } from './types';
+
+import { FleetFromHostFilesClient } from './client_from_host';
+import type { FileCustomMeta } from './types';
+import type { FleetFileUpdatableFields, HapiReadableStream } from './types';
+
+export class FleetToHostFilesClient
+  extends FleetFromHostFilesClient
+  implements FleetToHostFileClientInterface
+{
+  constructor(
+    esClient: ElasticsearchClient,
+    logger: Logger,
+    packageName: string,
+    maxSizeBytes?: number
+  ) {
+    super(esClient, logger, packageName);
+
+    // FIXME:PT define once we have new index patterns (defend workflows team issue #6553)
+    this.fileMetaIndex = getFileMetadataIndexName(packageName);
+    this.fileDataIndex = getFileDataIndexName(packageName);
+
+    this.esFileClient = createEsFileClient({
+      metadataIndex: this.fileMetaIndex,
+      blobStorageIndex: this.fileDataIndex,
+      elasticsearchClient: esClient,
+      logger,
+      indexIsAlias: true,
+      maxSizeBytes,
+    });
```

Comment:
i'm not sure it's okay that `createEsFileClient()` is called twice, first from `super()` then from this constructor. what do you think?

maybe there could be a base class (implementing the shared `get()`, `doesFileHaveData()` and `download()`) that could have a simplified constructor, and the two child classes (`FromHost` and `ToHost`) can have their own constructors without conflicting with each other.

or, the constructors could be `private` and `createInstance()` functions implemented in each class can create the instance with the correct `esFileClient` with a simplified constructor

Reply:
There is no issue, but I agree its a bit strange to create one instance and then overwrite it with another. I think I will instead change the constructor to instead take in the ES file client as a required dependency.

I did not want to create yet another "base class" just for the few methods.

Reply:
I ended up changing the base class (`*FromHost`) to take as input the index names instead of the package name.

Reply:
nice idea! 👏  thanks for the modification!